src/compiler/utilities.ts

```diff
@@ -3996,16 +4016,32 @@ namespace ts {
 
         function writeLine(force?: boolean) {
             if (!lineStart || force) {
-                output += newLine;
-                lineCount++;
-                linePos = output.length;
-                lineStart = true;
+                appendAndUpdateLineCountAndPos(newLine);

```

Comment:
Bonus failure: the tests pass empty string as `newLine` and depend on this to mark the next position as a line-start regardless.